### ISSUE TITLE
Failing under 3.0.3

### ISSUE THREAD

**Issue by gmerritt123** [carve11/bokeh#1]
First of all this is really, really well done. Thank you so much for making this.

Unfortunately, there seems to be some sort of font recognition issue when applying in bokeh 3.0.3:

![image](https://user-images.githubusercontent.com/58000181/210416908-0253f28f-4937-4240-b2d2-8223a6567e18.png)

I did some cursory playing around and googling and was not able to identify anything obvious. I tried changing to 'arial' from 'helvetica' for example but no dice. I feel like it may potentially be an upstream issue (i.e. on bokeh's dependencies' end)... but as sole author of this extension you may be able to identify/diagnose. Thanks again...

**Comment by carve11** [comment]
Thanks for your comment. I'll look into this.

**Comment by gmerritt123** [comment]
I'll share my mercator workaround/alternative, you may find it interesting/useful to exploit. It leverages the (new to 3.x) CustomJSTicker. The idea is to:

- initialize a empty FixedTicker and assign to the axis
- use a CustomJS callback, that triggers on xrange/yrange update that :
    -  translates the webmercator x/y ranges to lat long;
    -  calculates the lat-long distance across the x range using haversine formula;
    - then with that distance, use d3.ticks to get nice rounded tick values across those distances; 
    - convert those tick values back to their web mercator values (using a basic ratio idea); and
    - set that result to the FixedTicker's ticks
- Then create a CustomJSTicker that translates those ticks back into their distance equivalent

The python:

```
def setMercatorDistanceAxis(pv,axis='x'):
    '''function to update a plan view/map figure's x axis ticks to dynamically display distance instead of webmercator coords'''
    from bokeh.models import CustomJSTickFormatter, FixedTicker
    tk = FixedTicker()
    pv.xaxis.ticker = tk   
    cb = CustomJS(args=dict(yr=pv.y_range,xr=pv.x_range,tk=tk,xa=pv.xaxis[0])
                  ,code='''
                    var o1 = webMercToLatLong({'x':xr.start,'y':yr.start})
                    var om = webMercToLatLong({'x':(xr.start+xr.end)/2,'y':yr.start})
                    var o2 = webMercToLatLong({'x':xr.end,'y':yr.start})                                        
                    var d1 = latlongDistance(o1,om)
                    var d2 = latlongDistance(o2,om)
                    var d = d1+d2
                    var tks = d3.ticks(0,d,6)
                    var interp = d3.interpolateNumberArray([0,xr.start],[d,xr.end])
                    var upd_tks = tks.map(x=>interp(x/d)[1])
                    tk.ticks = upd_tks
                    if (d<10000){
                            xa.axis_label ='Distance (m)'
                            }
                    else {xa.axis_label = 'Distance (km)'}
                  ''')
    pv.y_range.js_on_change('start',cb)
    pv.y_range.js_on_change('end',cb)
    pv.x_range.js_on_change('start',cb)
    pv.x_range.js_on_change('end',cb)
    cjs_ticks = CustomJSTickFormatter(args=dict(yr=pv.y_range,xr=pv.x_range)
                                              ,code='''
                                               var o1 = webMercToLatLong({'x':xr.start,'y':yr.start})
                                               var om = webMercToLatLong({'x':(xr.start+xr.end)/2,'y':yr.start})
                                               var o2 = webMercToLatLong({'x':xr.end,'y':yr.start})                                        
                                               var d1 = latlongDistance(o1,om)
                                               var d2 = latlongDistance(o2,om)
                                               var d = d1+d2
                                               var r = d/(xr.end-xr.start)      
                                               if (d<10000){
                                                       return Math.round((tick-xr.start)*r)
                                                       }
                                               else {return Math.round((tick-xr.start)*r/1000)}
                                               ''')
    pv.xaxis.formatter = cjs_ticks
    print("Figure's X Axis updated to display distance")
```

The JS functions used:
```
//converts webmerc {'x':,'y':} to lat long {'lat':,'lon'}
function webMercToLatLong(p){
  const k = 6378137 
  var lon = p.x / (k*Math.PI/180)
  var lat = Math.atan(Math.exp(p.y/k))/(Math.PI/360)-90
  return {'y':lat,'x':lon}
  }

//calcs distance between two lat longs {'y':,'x'}
function latlongDistance(p1, p2) {
  var p = 0.017453292519943295;    // Math.PI / 180
  var c = Math.cos;
  var a = 0.5 - c((p2.y - p1.y) * p)/2 + 
		  c(p1.y * p) * c(p2.y * p) * 
		  (1 - c((p2.x - p1.x) * p))/2;

  return 1000* 12742 * Math.asin(Math.sqrt(a)); // 2 * R; R = 6371 km
	}
```

I also use d3 ( https://cdnjs.cloudflare.com/ajax/libs/d3/7.8.0/d3.min.js ) as a resource, but I think there may be bokeh-built in equivalent to d3.ticks and d3.interpolateNumberArray that may make it unnecessary.

![dynoscale](https://user-images.githubusercontent.com/58000181/210554481-67405ea5-74ea-449a-9cec-920e10fce401.gif)

**Comment by carve11** [comment]
I have now updated the code. It was a minor bug with `TextBox` and getting the width. But I chose to rewrite most of the code (not fluent in typescript).
Thanks for sharing the code above. I would assume Bokehs `AdaptiveTicker` gives nice ticks when you are using an axis. The reason for me not using this ticker is I was not able to get the ticks I would like with respect to multiples. When I also have a size constraint on the scale bar I coded my own function to get nice values for the ticks.
